source/data-modeling/persistence-configuration.txt

```diff
@@ -0,0 +1,296 @@
+Perhaps you want to store your model's documents in a collection with a 
+different name than the pluralized form of the model class name.
+
+You can use the ``store_in`` macro to change the collection, database, or client
+where {+odm+} stores a model's documents. The following example shows how
+to use the ``store_in`` macro to store documents from the ``Person`` class in
+a collection called ``citizens`` in the ``other`` database within a client
+named ``analytics``.
+
+.. literalinclude:: /includes/data-modeling/persistence-configuration.rb
+   :language: ruby
+   :start-after: start store_in example
+   :end-before: end store_in example
+
+The ``store_in`` macro can also accept a lambda. For example, you might want to 
+use this pattern in a multi-tenant application,
+where multiple users share common access to an application. You can use a lambda
+to perform operations on different databases based on the current thread so that
+users cannot access each others' data. The following example stores documents in
+a database determined by the current thread:
+
+.. literalinclude:: /includes/data-modeling/persistence-configuration.rb
+   :language: ruby
+   :start-after: start store_in lambda example
+   :end-before: end store_in lambda example
+
+.. _mongoid-set-runtime-persistence-options:
+
+Runtime Persistence Options
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can use the ``with()`` method on a model class or instance to change
+a model's persistence configuration for a group of operations during runtime.
+
+The persistence context for the model class or instance that you call ``with()``
+on changes to the defined temporary persistence context within a block. 
+When you call ``with()`` on a model class or instance, the class or instance is 
+yielded to the block. The temporary persistence context applies to both queries 
+and writes.
+
+By default, {+odm+} stores documents for the ``Band`` class in a collection called
+``bands``. The following code example uses the ``with()`` method to temporarily
+use a different client, database, and collection to perform operations on the 
+``Band`` class's documents:
+
+.. literalinclude:: /includes/data-modeling/persistence-configuration.rb
+   :language: ruby
+   :start-after: start with() example
+   :end-before: end with() example
+
+.. note:: Define Clients
+
+   In the preceding example, you must define the ``tertiary`` cluster under
+   ``clients`` in your ``mongoid.yml`` file.
+
+Ensuring Consistency in Contexts
+++++++++++++++++++++++++++++++++
+
+If you save a document in a temporary persistence context, it does not exist
+in the default persistence context if you do not save it there as well. 
+For example, if you insert a ``Band`` model document into the ``artists`` 
+collection, the same document will not be inserted into the ``bands`` collection.
+Ensure that you perform operations on the same persistence context for data
+consistency. The following example uses ``with()`` to insert a document into
```

Comment:
Q: does this sentence mean 'operations work only on one context at a time, so if you're working on more than one, make sure you don't mix them up'?